faucet/src/faucet/simple_faucet.rs

```diff
@@ -0,0 +1,187 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::sync::Arc;
+
+use anyhow::anyhow;
+use async_trait::async_trait;
+use futures::lock::Mutex;
+use sui::wallet_commands::WalletContext;
+use sui_types::{
+    base_types::{ObjectID, SuiAddress},
+    gas_coin::GasCoin,
+    messages::{ExecutionStatus, Transaction},
+    object::Object,
+};
+use tracing::info;
+
+use crate::{Faucet, FaucetError, FaucetReceipt};
+
+/// A naive implementation of a faucet that processes
+/// request sequentially
+pub struct SimpleFaucet {
+    // TODO: handle concurrency correctly
+    wallet: Arc<Mutex<WalletContext>>,
```

Comment:
I am not sure if I should use `Arc<Mutex<>>` here as well cc @patrickkuo

Reply:
Don't need Arc Mutex anymore because the GatewayApi is now immutable